releasenotes/notes/gate_param_validation-76cd50fc4ab75293.yaml

```diff
@@ -0,0 +1,5 @@
+---
+deprecations:
+  - |
+    The use of `numpy.ndarray` as a parameter type for the class
+    :class:`~qiskit.circuit.gate.Gate` is being deprecated.
```

Comment:
This was only part of what I was talking about in #5099. It is more than just `numpy.ndarray` that is deprecated. #4181 changed the entire mechanism for Gate subclasses to set parameters. This change needs to be documented in a way that tells users who may have custom gates defined (which in this case includes aer) how to migrate from a previously supported method to something that works with this new model. Right now this doesn't describe how a user can adapt to the deprecation or the new model.

Also are we really deprecating an `ndarray` as a parameter that seems incorrect to do, especially since it's a valid type on instruction parameter: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/circuit/instruction.py#L59

Reply:
I agree that `Instruction` should not accept anything. In this PR I'm considering `Instruction` as abstract.